tests/h/presenters/annotation_json_test.py

```diff
@@ -1,208 +1,119 @@
+            get_formatter({"nipsa": "maybe"}),
+        ]
 
-        presented = presenter.asdict()
+        presented = AnnotationJSONPresenter(context, formatters=formatters).asdict()
 
         assert presented["flagged"] == "nope"
         assert presented["nipsa"] == "maybe"
 
-    def test_immutable_formatters(
-        self, groupfinder_service, links_service, get_formatter
-    ):
-        """Double-check we can't mutate the formatters list after the fact.
-
-        This is an extra check just to make sure we can't accidentally change
-        the constructor so that it simply aliases the list that's passed in,
-        leaving us open to all kinds of mutability horrors.
-
-        """
-        ann = mock.Mock(id="the-real-id", extra={})
-        context = AnnotationContext(ann, groupfinder_service, links_service)
-        formatters = []
+        for formatter in formatters:
+            formatter.format.assert_called_once_with(context)
 
-        presenter = AnnotationJSONPresenter(context, formatters)
-        formatters.append(get_formatter({"enterprise": "synergy"}))
-        presented = presenter.asdict()
-
-        assert "enterprise" not in presented
-
-    def test_formatter_uses_annotation_context(
-        self, groupfinder_service, links_service, get_formatter
-    ):
-        annotation = mock.Mock(id="the-id", extra={})
-        context = AnnotationContext(annotation, groupfinder_service, links_service)
-        formatter = get_formatter()
-
-        presenter = AnnotationJSONPresenter(context, formatters=[formatter])
-        presenter.asdict()
-
-        formatter.format.assert_called_once_with(context)
-
-    @pytest.mark.usefixtures("policy")
     @pytest.mark.parametrize(
-        "annotation,group_readable,action,expected",
-        [
-            (
-                mock.Mock(userid="acct:luke", shared=False),
-                "world",
-                "read",
-                ["acct:luke"],
-            ),
-            (
-                mock.Mock(userid="acct:luke", groupid="abcde", shared=False),
-                "members",
-                "read",
-                ["acct:luke"],
-            ),
-            (
-                mock.Mock(groupid="__world__", shared=True),
-                "world",
-                "read",
-                ["group:__world__"],
-            ),
-            (
-                mock.Mock(groupid="lulapalooza", shared=True),
-                "members",
-                "read",
-                ["group:lulapalooza"],
-            ),
-            (
-                mock.Mock(groupid="open", shared=True),
-                "world",
-                "read",
-                ["group:__world__"],
-            ),
-            (mock.Mock(userid="acct:luke"), None, "admin", ["acct:luke"]),
-            (mock.Mock(userid="acct:luke"), None, "update", ["acct:luke"]),
-            (mock.Mock(userid="acct:luke"), None, "delete", ["acct:luke"]),
-        ],
```

Comment:
This is the major motivator for looking at these tests. What the flipping heck is this up to?

To me this looks like a duplication of the testing done for the `AnnotationContext`. Note that in the previous version the real `AnnotationContext` was used, not a mocked one.

Reply:
Yeah it's quite hard to follow. The new much smaller test is much better :+1: